### PR TITLE
editor: fix field visibility on the model change

### PR DESCRIPTION
**Pull request by jma**
- Adds new onPopulate hooks.
- Shows a hidden field when a value exists in the model.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- To fix the fields visibility on model changes.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 1]
```diff
@@ -287,6 +274,34 @@ export class EditorComponent implements OnInit, OnDestroy {
...
+    }
+    model = removeEmptyValues(model);
+    const modelEmpty = isEmpty(model);
+    if (!modelEmpty && (field.hide !== false)) {
+      setTimeout(() => {
+        field.hide = false;
+        this.editorService.removeHiddenField(field);
+      });
+    }
+    if (modelEmpty && (field.templateOptions.hide === true && field.hide === undefined)) {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts]: Why testing *field.hide === undefined* ? It's probably correct but I don't understand why..

File: projects/rero/ng-core/src/lib/record/editor/editor.service.ts [hunk 2]
```diff
@@ -59,21 +59,34 @@ export class EditorService {
     this._hiddenFieldsSubject.next(this._hiddenFields);
   }
 
+  /**
+   * Check if the field is a candidate for the field navigation.
+   * @param field - FormlyFieldConfig, form config to be added
+   * @returns boolean, true if the field is at the root JSONSchema.
+   */
+  isFieldRoot(field: FormlyFieldConfig) {
+    return field.parent.parent.parent === undefined;
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/editor.service.ts]: Is there a possibility that field.parent.parent is undefined and crash this condition?

Comment (jma) [reply projects/rero/ng-core/src/lib/record/editor/editor.service.ts]: Ad I understood no, but we can add some checks.

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts (hunk 1 above)
Comment (jma) [reply projects/rero/ng-core/src/lib/record/editor/editor.component.ts]: at the Initial step hide is undefined means untouched

**Pull request merged**